File: js_implemention/bubble_sort.js
function bubble_sort(arr, n) {
    let temp;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

function execution() {
    arr = [3, 8, 12, 78, 2, 4];
    n = 6;
    bubble_sort(arr, n);
    // print the sorted array 
    for (let e of arr) {
        console.log(e);
    }
}
execution();

File: c++_implemention/bubble_sort.cpp
#include<bits/stdc++.h>
using namespace std;
#define doura ios::sync_with_stdio(0);cin.tie(0)
#define ll long long
#define sesh '\n';

void bubble_sort(int arr[], int n) {
    int temp;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}



int main() {

    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    bubble_sort(arr, n);
    // let's print the sorted array
    for (auto e : arr) {
        cout << e << ' ';
    }
    
    return 0;
}
